### ISSUE TITLE
[3.0] MeshDataTool returns incorrect normals

### ISSUE THREAD

**Issue by supagu** [godotengine/godot#8771]
MeshDataTool is returning incorrect normals in godot 3. Godot 2 works correctly.

I have a sample project here which demonstrates the problem:
https://drive.google.com/open?id=0B_nQZvJoqbFmQzQ5TmptNDU4OTg

Open Test-Cube.tscn in godot 3 and run the scene and it spits out to console:

```
normal[0] = (-1, 0.011765, -1)
normal[1] = (-1, 0.011765, -1)
normal[2] = (-1, 0.011765, -1)
normal[3] = (-1, -0.003922, -1)
normal[4] = (-1, -0.003922, -1)
normal[5] = (-1, -0.003922, -1)
normal[6] = (-0.003922, -1, -1)
normal[7] = (-0.003922, -1, -1)
normal[8] = (-0.003922, -1, -1)
normal[9] = (-1, -1, -0.003922)
normal[10] = (-1, -1, -0.003922)
normal[11] = (-1, -1, -0.003922)
normal[12] = (0.011765, -1, -1)
normal[13] = (0.011765, -1, -1)
normal[14] = (0.011765, -1, -1)
normal[15] = (-1, -1, 0.011765)
normal[16] = (-1, -1, 0.011765)
normal[17] = (-1, -1, 0.011765)
normal[18] = (-1, 0.011765, -1)
normal[19] = (-1, -0.003922, -1)
normal[20] = (-0.003922, -1, -1)
normal[21] = (-1, -1, -0.003922)
normal[22] = (0.011765, -1, -1)
normal[23] = (-1, -1, 0.011765)
```

open Test-Cube-2.tscn in godot 2 and it outputs to console:

```
normal[0] = (0, -1, 0)
normal[1] = (0, -1, 0)
normal[2] = (0, -1, 0)
normal[3] = (0, -1, 0)
normal[4] = (0, 1, 0)
normal[5] = (0, 1, 0)
normal[6] = (0, 1, 0)
normal[7] = (0, 1, 0)
normal[8] = (1, 0, 0)
normal[9] = (1, 0, 0)
normal[10] = (1, 0, 0)
normal[11] = (1, 0, 0)
normal[12] = (-0, -0, 1)
normal[13] = (-0, -0, 1)
normal[14] = (-0, -0, 1)
normal[15] = (-0, -0, 1)
normal[16] = (-1, -0, -0)
normal[17] = (-1, -0, -0)
normal[18] = (-1, -0, -0)
normal[19] = (-1, -0, -0)
normal[20] = (0, 0, -1)
normal[21] = (0, 0, -1)
normal[22] = (0, 0, -1)
normal[23] = (0, 0, -1) 
```

I have a hunch that if reading normals is wrong, writing them may also be wrong.

**Comment by supagu** [comment]
Checked again on 15 june, 2017. Still an issue.

**Comment by supagu** [comment]
Resolved in #9230
